sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/util/CosmosPagedFlux.java

```diff
@@ -125,11 +139,103 @@ public void subscribe(CoreSubscriber<? super T> coreSubscriber) {
                 pagedFluxOptions.getTracerProvider().endSpan(parentContext.get(), Signal.error(throwable),
                     TracerProvider.ERROR_CODE);
             }
+
+            if (pagedFluxOptions.getCosmosAsyncClient() != null &&
+                Configs.isClientTelemetryEnabled(BridgeInternal.isClientTelemetryEnabled(pagedFluxOptions.getCosmosAsyncClient())) &&
+                throwable instanceof CosmosException) {
+                CosmosException cosmosException = (CosmosException) throwable;
+                fillClientTelemetry(pagedFluxOptions.getCosmosAsyncClient(), 0, pagedFluxOptions.getContainerId(),
+                    pagedFluxOptions.getDatabaseId(),
+                    pagedFluxOptions.getOperationType(), pagedFluxOptions.getResourceType(),
+                    BridgeInternal.getContextClient(pagedFluxOptions.getCosmosAsyncClient()).getConsistencyLevel(),
+                    (float) cosmosException.getRequestCharge(), Duration.between(startTime.get(), Instant.now()));
```

Comment:
I wonder why we can't keep it double in the client telemetry as well ?

Reply:
We want to keep the size as low as possible, float give 6 to 7 place of decimal value , more than enough for us to measure request charge.